src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step/lib/for_each_field.js

```diff
@@ -0,0 +1,58 @@
+import _ from 'lodash';
+import isGeoPointObject from './is_geo_point_object';
+
+// This function recursively traverses an object, visiting each node that elasticsearch would index as a field.
+// Iteratee is invoked with two arguments: (value, fieldName). fieldName is the name of the field as elasticsearch
+// would see it. For example:
+//
+// const testDoc = {
+//   foo: [
+//     {bar: [{'baz': 1}]},
+//     {bat: 'boo'}
+//   ],
+//   geo: {
+//     lat: 38.6631,
+//     lon: -90.5771
+//   }
+// };
+//
+// forEachField(testDoc, function(value, fieldName) { ... });
+//
+// The iteratee would be invoked six times, with the following parameters:
+// 1. fieldName = 'foo'          value = {bar: [{'baz': 1}]}
+// 2. fieldName = 'foo'          value = {bat: 'boo'}
+// 3. fieldName = 'foo.bar'      value = {'baz': 1}
+// 4. fieldName = 'foo.bar.baz'  value = 1
+// 5. fieldName = 'foo.bat'      value = 'boo'
+// 6. fieldName = 'geo'          value = {lat: 38.6631, lon: -90.5771}
+//
+// forEachField handles arrays, objects, and geo_points as elasticsearch would. It does not currently handle nested
+// type fields.
+
+function forEachFieldAux(value, iteratee, fieldName) {
+  if ((!_.isObject(value) && !_.isArray(value)) || isGeoPointObject(value)) {
```

Comment:
can't `(!_.isObject(value) && !_.isArray(value))` be reduced to `!_.isObject(value)`